codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -523,36 +530,47 @@ public void operationComplete(ChannelFuture future) throws Exception {
      * Sends a {@code GO_AWAY} frame to the remote endpoint and updates the connection state appropriately.
      */
     @Override
-    public ChannelFuture writeGoAway(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData,
-            ChannelPromise promise) {
-        Http2Connection connection = connection();
-        if (connection.goAwayReceived() || connection.goAwaySent()) {
+    public ChannelFuture writeGoAway(final ChannelHandlerContext ctx, final int lastStreamId, final long errorCode,
+                                     final ByteBuf debugData, ChannelPromise promise) {
+        try {
+            if (connection().goAwaySent() && connection().remote().lastKnownStream() < lastStreamId) {
```

Comment:
`connection()` is used a few times in this method.  Lets save in a local variable and re-use.